Source/ProgressionSystemRuntime/Public/Data/PSWorldSubsystem.h

```diff
@@ -170,8 +178,16 @@ class PROGRESSIONSYSTEMRUNTIME_API UPSWorldSubsystem : public UWorldSubsystem
 	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
 	void OnGameStateChanged(ECurrentGameState CurrentGameState);
 
+	/** Updates the chosen player mesh on the level and switch current save game row */
+	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void ChangeCurrentPlayer(UPSSpotComponent* SpotComponent);
+	
 	/** Checks if the current character is unlocked and the player is allowed to play, and if not allowed, sets the previous character. */
 	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
 	void CheckAndSetCharacterUnlockStatus();
+
+	/** Switches back to the last spot character in case of player tried to play with locked character as client */
+	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void SwitchToLastSpotCharacter();
```

Comment:
Per [d179119](https://github.com/JanSeliv/Bomber/commit/d179119c6dad03ab3bb0d081c26ae1714d392b94), this bug is fixed now. See the video: [link](https://github.com/user-attachments/assets/c1f4cac9-0094-47be-a0ab-4d8792205a4f).

The NMM module now handles the logic of switching the blocked character to the previous one on its own, so please:
1. Remove unnecessary logic from this review, such as caching spots and the `SwitchToLastSpotCharacter` function.
2. Don't call `SetCustomPlayerMeshData` from the PS module anymore, as it's now the responsibility of the NMM module.

Reply:
removed unnecessary logic as par of https://github.com/h4rdmol/ProgressionSystem/pull/3/commits/eded2e446ff1204eb63d3800bbcce887c41fa82a